certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/unit/service/GetUnitMessagesDomainService.java

```diff
@@ -0,0 +1,51 @@
+package se.inera.intyg.certificateservice.domain.unit.service;
+
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import se.inera.intyg.certificateservice.domain.action.model.ActionEvaluation;
+import se.inera.intyg.certificateservice.domain.action.model.CertificateActionType;
+import se.inera.intyg.certificateservice.domain.certificate.repository.CertificateRepository;
+import se.inera.intyg.certificateservice.domain.common.model.MessagesRequest;
+import se.inera.intyg.certificateservice.domain.common.model.MessagesResponse;
+import se.inera.intyg.certificateservice.domain.message.model.Message;
+import se.inera.intyg.certificateservice.domain.message.model.MessageType;
+import se.inera.intyg.certificateservice.domain.message.repository.MessageRepository;
+
+@RequiredArgsConstructor
+public class GetUnitMessagesDomainService {
+
+  private final MessageRepository messageRepository;
+  private final CertificateRepository certificateRepository;
+
+  public MessagesResponse get(MessagesRequest request, ActionEvaluation actionEvaluation) {
+    final var messages = messageRepository.findByMessagesRequest(request)
+        .stream()
+        .filter(message -> !isReminderOrAnswer(message))
+        .toList();
+
+    final var certificates = messages.stream()
+        .map(Message::certificateId)
+        .distinct()
+        .map(certificateRepository::getById)
+        .filter(certificate -> certificate.allowTo(
+                CertificateActionType.READ,
+                Optional.of(actionEvaluation)
+            )
+        ).toList();
+
+    final var filteredMessagesBasedOnActionEvaluation = messages.stream()
```

Comment:
I dont really understand why this is needed, since the search we make on line 26 should assert that these are the same?

Maybe an update of the variable name could make the purpose clearer?

Reply:
1. Get all messages
2. Get all certificates
3. Filter certificates which user is not allowed to read
4. Filter messages that user is not allowed to read (the messages that belong to a certificate which has been filtered). I think the name "filteredMessagesBasedOnActionEvaluation" describes this.

Reply:
Ah, I see, very good. Thats something that could be easily missed, great that u catched it! 

To me it would be easier to understand if we updated the variable to ```filteredMessagesBasedOnAllowedToRead```, but thats subjective and im fine with you keeping it as it. 

Thanks for clarifying!